clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -769,7 +769,7 @@ boolean isEndTxn() {
         }
 
         @Override
-        public void handleResponse(AbstractResponse response) {
+        public synchronized void handleResponse(AbstractResponse response) {
```

Comment:
This and the other handlers should synchronize on `TransactionManager.this`, right?